packages/@ember/-internals/runtime/tests/system/object/create_test.js

```diff
@@ -219,5 +220,22 @@ moduleFor(
         },
       }).create(options);
     }
+
+    ['@test does not create enumerable properties for owner and init factory when created by the container'](
+      assert
+    ) {
+      let registry = new Registry();
+      let container = registry.container();
+      container.owner = {};
+
+      registry.register('component:foo-bar', EmberObject);
+
+      let componentFactory = container.factoryFor('component:foo-bar');
+      let instance = componentFactory.create();
+
+      assert.deepEqual(Object.keys(instance), [], 'no enumerable properties were added');
+      assert.equal(getOwner(instance), container.owner, 'owner was defined on the instance');
+      assert.ok(getFactoryFor(instance), 'factory was defined on the instance');
+    }
```

Comment:
This test is great (and is exactly what I was hoping for), can you make another one (or more assertions) that use `container.lookup` also?